airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -145,20 +165,26 @@ public void run(final ConnectionUpdaterInput connectionUpdaterInput) throws Retr
 
               final StandardSyncSummary standardSyncSummary = standardSyncOutput.get().getStandardSyncSummary();
 
+              // if the current workflow was a reset, it is now safe to switch it back to "normal".
               if (workflowState.isResetConnection()) {
                 workflowState.setResetConnection(false);
               }
 
               if (standardSyncSummary != null && standardSyncSummary.getStatus() == ReplicationStatus.FAILED) {
+                // todo (cgardens) - why are failures a member field as opposed to a local variable? seems like it
+                // is just asking for us to make a mistake when resetting it.
                 failures.addAll(standardSyncOutput.get().getFailures());
+                // todo (cgardens) - why is it a member field?
                 partialSuccess = standardSyncSummary.getTotalStats().getRecordsCommitted() > 0;
                 workflowState.setFailed(true);
               }
             } catch (final ChildWorkflowFailure childWorkflowFailure) {
               if (childWorkflowFailure.getCause() instanceof CanceledFailure) {
                 // do nothing, cancellation handled by cancellationScope
+                // todo (cgardens) - what does this ^ mean?
 
               } else if (childWorkflowFailure.getCause() instanceof ActivityFailure) {
+                // todo (cgardens) - what is this?
```

Comment:
Let's check with Parker

Reply:
This ensures that we can properly classify unhandled failures from the sync workflow. If the normalization activity throws an exception, for example, this lets us set the failureOrigin to `normalization`.